### PR TITLE
integration tests: faster startup time and add clear.sh helper

### PR DESCRIPTION
**Pull request by listx**
/assign @cjwagner 
/cc @chaodaiG @mpherman2

### CODE REVIEW

**Comment by chaodaiG** [comment]
based on offline discussed, part of the purpose of this script is for cleaning up prowjobs CRs from horologium, the root cause of this problem is that the predefined periodic prowjob keeps on scheduling prow jobs every minute, and this script will help cleaning them up during local development. This is good as a temporary solution, and as a longer term solution, this is better cleaned up by the horologium test itself, which could be done by:
- no prowjob is defined at integration test set up phase.
- horologium test add the prowjob config  before it starts.
- horologium test cleans up everything it launched afterwards.

@listx , is there anything else that needs to be cleaned up?

**Comment by listx** [comment]
> based on offline discussed, part of the purpose of this script is for cleaning up prowjobs CRs from horologium, the root cause of this problem is that the predefined periodic prowjob keeps on scheduling prow jobs every minute, and this script will help cleaning them up during local development. This is good as a temporary solution, and as a longer term solution, this is better cleaned up by the horologium test itself, which could be done by:
> 
> * no prowjob is defined at integration test set up phase.
> * horologium test add the prowjob config  before it starts.
> * horologium test cleans up everything it launched afterwards.
> 
> @listx , is there anything else that needs to be cleaned up?

That's a good summary. I can't think of anything else at the moment.

**Comment by chaodaiG** [comment]
Would you mind add a comment like that somewhere so we won't forget to remove this cleanup.sh once the longer term solution is done?

**Comment by listx** [comment]
> Would you mind add a comment like that somewhere so we won't forget to remove this cleanup.sh once the longer term solution is done?

Thinking about this again, I think it is still useful to have this script around. One thing that I constantly fear is typing in the wrong kubectl context when deleting this and using this script guarantees that we always delete resources from the test cluster, not production. There's value in that when writing new tests and the tests aren't always completing successfully such that the test resources are not getting cleaned up.

Also I added more speedup improvements in this PR. PTAL

File: prow/test/integration/clear.sh [hunk 1]
```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env bash
+# Copyright 2021 The Kubernetes Authors.
```
Comment (chaodaiG) [inline prow/test/integration/clear.sh:2]: 2022

**Review by chaodaiG — APPROVED** [review]
have a nit but not important at all imo

**Pull request merged**